neurom/fst/_neuronfunc.py

```diff
@@ -133,3 +135,68 @@ def _elevation(section, soma):
     return [_elevation(s.root_node.points, n.soma)
             for n in nrns
             for s in n.neurites if neurite_filter(s)]
+
+
+def sholl_crossings(neurites, center, radii):
+    '''calculate crossings of neurites
+
+    Args:
+        nrn(morph): morphology on which to perform Sholl analysis
+        radii(iterable of floats): radii for which crossings will be counted
+
+    Returns:
+        Array of same length as radii, with a count of the number of crossings
+        for the respective radius
+    '''
+    def _count_crossings(neurite, radius):
+        '''count_crossings of segments in neurite with radius'''
+        r2 = radius ** 2
+        count = 0
+        for start, end in iter_segments(neurite):
+            start_dist2, end_dist2 = (morphmath.point_dist2(center, start),
+                                      morphmath.point_dist2(center, end))
+            count += int(start_dist2 <= r2 <= end_dist2 or
+                         end_dist2 <= r2 <= start_dist2)
```

Comment:
My impression is that this counts the crossings of each segment with each circle. The only problem here is what happens if the section would bend back. In this case we would have two crossings (for two segments) for the same section. This might be rare and even unimportant but I think it will worth to note it in the documentation?